test/test_reductions.py

```diff
@@ -2257,17 +2257,49 @@ def test_against_np(tensor, bins=100, min=0, max=0):
         expanded = torch.randn(1, 5, 1, 2, device=device).expand(3, 5, 7, 2)
         test_against_np(expanded)
 
-    def test_reduction_empty(self, device):
```

Comment:
Can we get a cliffs notes on what changed here?

Reply:
The previous version did perform error checking in terms of the new error messages, and all the functions under the reduction domain were clubbed under a single test. This separates them out in terms of commonality of testing. For example, the `any` and `all` tests don't require sending a particular dtype. Also, there is no need to check for 'identity' anymore as was being done in the previous tests since the functions that such a check are now under a different test.

Reply:
Sorry for being late to the party here, this was buried under a pile of notifications.

A couple things:

- let's add comments describing what the tests cover
- can we compare with NumPy to be sure our reduction behavior is the same? for example, for most of these operations these tuples could be extended with the NumPy operation to compare again. It's OK if not every operator has a NumPy counterpart, and the NumPy comparison part of the test can be skipped in that case. Ping me if you have questions about this suggestion, @v0dro 

cc @heitorschueroff this is the type of test we should look to automating with reduction-specific opinfos

Reply:
I've added tests for both points raised by @mruberry .